plugins/cpu/fu-cpu-device.c

```diff
@@ -418,6 +425,90 @@ fu_cpu_device_add_security_attrs_intel_tme(FuCpuDevice *self, FuSecurityAttrs *a
 	fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS);
 }
 
+static void
+fu_cpu_device_add_security_attrs_amd_entry_sign(FuCpuDevice *self, FuSecurityAttrs *attrs)
+{
+	FuContext *ctx = fu_device_get_context(FU_DEVICE(self));
+	const gchar *agesa_stream;
+	const gchar *version;
+	guint64 ucode_version = 0;
+	guint32 ucode_version_fixed;
+	g_autoptr(FwupdSecurityAttr) attr = NULL;
+	g_autoptr(GError) error_local = NULL;
+	g_autoptr(GHashTable) cpu_attrs = NULL;
+	g_auto(GStrv) split = NULL;
+
+	/* create attr */
+	attr = fu_device_security_attr_new(FU_DEVICE(self), FWUPD_SECURITY_ATTR_ID_AMD_ENTRY_SIGN);
+	fwupd_security_attr_set_result_success(attr, FWUPD_SECURITY_ATTR_RESULT_LOCKED);
+	fu_security_attrs_append(attrs, attr);
+
+	/* get the AGESA stream e.g. `AGESA!V9 StrixKrackanPI-FP8 1.1.0.0a` */
+	agesa_stream = fu_context_get_smbios_string(ctx,
+						    FU_CPU_AGESA_SMBIOS_TYPE,
+						    FU_CPU_AGESA_SMBIOS_LENGTH,
+						    FU_CPU_AGESA_SMBIOS_OFFSET,
+						    &error_local);
+	if (agesa_stream == NULL) {
+		g_debug("cannot get AGESA stream: %s", error_local->message);
+		fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_MISSING_DATA);
+		return;
+	}
+	split = g_strsplit(agesa_stream, " ", 3);
+	if (g_strv_length(split) != 3) {
+		g_warning("invalid agesa stream format: %s", agesa_stream);
+		fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_MISSING_DATA);
+		return;
+	}
+	version = fu_cpu_stream_name_to_amd_entry_sign_agesa_version(split[1]);
+	if (version == NULL) {
+		g_warning("unknown agesa stream name: %s", split[1]);
+		fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_MISSING_DATA);
+		return;
+	}
+
+	/* the firmware is new enough to be fixed */
+	if (fu_version_compare(version, split[2], FWUPD_VERSION_FORMAT_UNKNOWN) < 0) {
+		g_debug("%s > %s, so good enough", version, split[2]);
+		fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS);
+		return;
+	}
+
+	/* is this fixed by microcode */
```

Comment:
I would argue this comment should say _mitigated_.  The reason I say this is because you still have an "open window" of time that you could have been attacked before the kernel loaded.

Reply:
Ack, changed.